test.py

```diff
@@ -0,0 +1,247 @@
+        # Create directory for APIDevKit
+        # Download APIDevKit
+        for version in acVersionList:
+            if version in configData['devKitLinks']:
+
+                devKitFolder = rootFolder / f'APIDevKit-{version}'
+                if not devKitFolder.exists():
+                    devKitFolder.mkdir()
+
+                devKitFolderList[version] = devKitFolder
+                DownloadAndUnzip(configData['devKitLinks'][version], devKitFolder, platformName)
+
+            else:
+                print('APIDevKit download link not provided')
+                return 1
+
+    # At this point, devKitFolderList dictionary has all provided ACVersions as keys
+    # For every ACVersion
+    # Enable execute permission if running on Mac
+    # If release, build Add-On for all languages with RelWithDebInfo configuration
+    # Else build Add-On with Debug and RelWithDebInfo configurations, without language specified   
+    # In each case, if package creation is enabled, copy the .apx/.bundle files to the Package directory
+    for version in devKitFolderList:
+        devKitFolder = devKitFolderList[version]
+
+        if platformName == 'MAC':
+            permissionParams = []
+            permissionParams.append ('chmod')
+            permissionParams.append ('+x')
+            permissionParams.append (str(devKitFolder / 'Support' / 'Tools' / 'OSX' / 'ResConv'))
+            permissionResult = subprocess.call (permissionParams)
+            if permissionResult != 0:
+                print ('Failed to grant permission')
+                return 1
+            
+        if args.release is True:
+            for languageCode in languageList:
+                if BuildAddOn(rootFolder, buildFolder, devKitFolder, addOnName, platformName, 'RelWithDebInfo', optionalParams, languageCode) != 0:
+                    return 1
+                if args.package:
+                    CopyResultToPackage(packageRootFolder, buildFolder, addOnName, platformName, 'RelWithDebInfo', languageCode, True)
+
+        else:
+            if BuildAddOn(rootFolder, buildFolder, devKitFolder, addOnName, platformName, 'Debug', optionalParams) != 0:
+                return 1
+            if BuildAddOn(rootFolder, buildFolder, devKitFolder, addOnName, platformName, 'RelWithDebInfo', optionalParams) != 0:
+                return 1
+            if args.package:
+                CopyResultToPackage(packageRootFolder, buildFolder, addOnName, platformName, 'Debug')
+                CopyResultToPackage(packageRootFolder, buildFolder, addOnName, platformName, 'RelWithDebInfo')
+
+    # Zip packages
+    if args.package:
+        release = 'Release_Localized' if args.release else 'Daily'
+        subprocess.call ([
+            '7z', 'a',
+            str(packageRootFolder.parent / (addOnName + '_' + release + '_' + platformName + '.zip')),
+            str(packageRootFolder / '*')
+        ])
+    
+    print ('Build Successful')
+    return 0
+
+sys.exit (Main ())

```

Comment:
Is there a reason that calling Main like this is neccessary?
if __name__ == "__main__":
    Main()
is the convention most of the times

Reply:
Fixed